kernels/ZendEngine3/fcall.c

```diff
@@ -283,8 +283,7 @@ int zephir_call_user_function(zval *object_pp, zend_class_entry *obj_ce, zephir_
 
 	zval p[fci.param_count];
 	for (i=0; i<fci.param_count; ++i) {
-		/** FIXME: ZVAL_COPY or ZVAL_COPY_VALUE? */
-		ZVAL_COPY(&p[i], params[i]);
+		ZVAL_COPY_VALUE(&p[i], params[i]);
```

Comment:
"Parameter 3 to preg_match_all() expected to be a reference, value given" - this was the clue